fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java

```diff
@@ -1694,6 +1694,29 @@ public Set<String> getVisiblePartitionNames() {
                 .collect(Collectors.toSet());
     }
 
+    public String getLatestHasDataPartitionValue() throws DdlException {
+        if (partitionInfo.getType() != PartitionType.RANGE) {
+            throw new DdlException("Currently only range partitions are supported");
```

Comment:
i would suggest implementing this function in `PartitionInfo`, so you can make specific implements for various types of partitions. We're going to make the list partition as default in the future, so only supporting range partition is not enough.

Reply:
Do you think we need to consider external table?

Reply:
> i would suggest implementing this function in `PartitionInfo`, so you can make specific implements for various types of partitions. We're going to make the list partition as default in the future, so only supporting range partition is not enough.

Good suggestion. I will directly reuse the PartitionInfo#getSortedPartition function, but this function does not yet have an implementation for EXPR_Range, so only LIST and RANGE partitions are currently supported. @murphyatwork